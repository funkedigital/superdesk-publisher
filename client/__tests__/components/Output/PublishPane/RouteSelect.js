import React from "react";
import RouteSelect from "../../../../components/Output/PublishPane/RouteSelect";
import { render } from "@testing-library/react";

const routes = [
  { id: 1, name: 'route1' },
  { id: 2, name: 'route2' },
  { id: 3, name: 'route3' }
];

describe("Output/PublishPane/RouteSelect", () => {
  it("renders properly", async () => {
    const { container, getByText } = render(
      <RouteSelect
        routes={routes}
        selectedRouteId={1}
        onChange={jest.fn()}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
