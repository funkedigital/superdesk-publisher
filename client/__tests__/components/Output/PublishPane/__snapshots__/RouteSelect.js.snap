// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/PublishPane/RouteSelect renders properly 1`] = `
<div
  class="sd-line-input sd-line-input--is-select sd-line-input--dark-ui sd-line-input--no-margin"
>
  <label
    class="sd-line-input__label"
  >
    Route
  </label>
  <select
    class="sd-line-input__select"
    name="routeId"
  >
    <optgroup
      label="Custom"
      style="color: black;"
    >
      <option
        value="1"
      >
        route1
      </option>
      <option
        value="2"
      >
        route2
      </option>
      <option
        value="3"
      >
        route3
      </option>
    </optgroup>
  </select>
</div>
`;
